### PR TITLE
feat: kickoff create new shimmer profile flow for mobile stardust

### LINKED ISSUE #4618: [Task]: Add welcome screen for onboarding
### Task description

Add welcome screen as router component to the mobile package and adjust the presentation.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4619: [Task]: Add protocol screen for onboarding
### Task description

Add protocol screen as router component to the mobile package and adjust the presentation.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4620: [Task]: Add network screen for onboarding
### Task description

Add network selection screen as router component to the mobile package and adjust the presentation.

Note: Private network doesn't have to be selectable.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4621: [Task]: Add shimmer profile setup screen for onboarding
### Task description

Add shimmer profile setup screen as router component to the mobile package and adjust the presentation.

Note: claim and restore do not have to be selectable.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4622: [Task]: Add create shimmer profile screen
### Task description

Add create shimmer profile screen as router component to the mobile package and adjust the presentation.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4623: [Task]: Add setup password screen for onboarding
### Task description

Add setup password screen as router component to the mobile package and adjust the presentation.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4624: [Task]: Add setup pin code screen for onboarding
### Task description

Add setup pin code screen as router component to the mobile package and adjust the presentation.

### Requirements

- [ ] Router component added to the mobile project
- [ ] Presentation adjusted for mobile

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4659: [Task]: Add onboarding layout
### Task description

Implement an onboarding layout that fits mobile requirements, especially in terms of keyboard interaction.

### Requirements

Mobile onboarding layout implemented

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by paul-boegelsack**
## Summary

This PR initializes the mobile project for stardust and implements the first iteration of onboarding views.

## Changelog

```
TBD
```

## Relevant Issues

Closes #4659
Closes #4618
Closes #4619
Closes #4620
Closes #4621
Closes #4622
Closes #4623
Closes #4624 

## Testing

Until the mobile bindings are ready, the `dev-mode/mobile-stardust` branch has to be utilized for testing:

1 `git checkout dev-mode/mobile-stardust` 
2 `git merge feat-mobile/new-shimmer-profile-ui`
3 `yarn (root directory)`
4 `yarn build (packages/desktop)`
5 `yarn start (packages/desktop)`

press `ctrl + d` to activate dev profile (It's necessary before selecting shimmer as protocol)
press `ctrl + c` to change the color theme
press `ctrl + k` to toggle the keyboard

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/mobile/routes/onboarding/views/storage-protection-setup/views/SetupPinProtectionView.svelte [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            pollLedgerNanoStatus()
+        }
+    })
+</script>
+
+<OnboardingLayout {onBackClick} {busy}>
+    <div slot="title">
+        <Text type="h2">{localize('views.onboarding.storageProtectionSetup.setupPinProtection.title')}</Text>
+    </div>
+    ´
```
Comment (begonaalvarezd) [inline packages/mobile/routes/onboarding/views/storage-protection-setup/views/SetupPinProtectionView.svelte:100]: there seems to be an extra character here

File: packages/mobile/tsconfig.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
             "@contexts/*": ["../shared/lib/contexts/*"],
             "@core/*": ["../shared/lib/core/*"],
             "@features/*": ["../shared/features/*"],
-            "@lib/*": ["../shared/lib/*"]
+            "@lib/*": ["../shared/lib/*"],
+            "@mobile/*": ["./*"]
```
Comment (begonaalvarezd) [inline packages/mobile/tsconfig.json:15]: I added an issue to make this work, I would remove it from here and try making it work in a different PR

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Wohoo amazing work man 🚀🚀 🚀 🚀 🚀 
Just added some comments and also waiting for a new update on this PR with the missing views, though I think they could perhaps be added in a different PR, up to you.
💪🏼 💪🏼 💪🏼 💪🏼

File: packages/mobile/components/OnboardingLayout.svelte [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        :global(h5) {
+            @apply font-bold;
+            @apply text-16;
+            @apply leading-140;
+        }
+    }
+    footer {
+        margin-bottom: env(safe-area-inset-bottom);
+    }
+    .highlight {
```
[the remark this answers is not in the document]
Comment (paul-boegelsack) [reply packages/mobile/components/OnboardingLayout.svelte]: Not needed as far as I can see, I removed it 👍

File: packages/mobile/routes/onboarding/views/storage-protection-setup/views/SetupPinProtectionView.svelte (hunk 1 above)
Comment (paul-boegelsack) [reply packages/mobile/routes/onboarding/views/storage-protection-setup/views/SetupPinProtectionView.svelte]: ups 😅

File: packages/mobile/components/OnboardingLayout.svelte [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        </div>
+    </div>
+    <footer style={$isKeyboardOpen && `margin-bottom: ${$keyboardHeight}px`}>
+        <slot name="footer" />
+    </footer>
+</div>
+
+<style type="text/scss">
+    header {
+        margin-top: env(safe-area-inset-top);
```
[the remark this answers is not in the document]
Comment (paul-boegelsack) [reply packages/mobile/components/OnboardingLayout.svelte]: Definitely, but I don't know if we need the safe areas for curved screens? Anyway, we can add them later, if needed, I guess.

File: packages/mobile/routes/onboarding/OnboardingRouter.svelte [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        NetworkSetupRouter,
+        ProfileSetupRouter,
+        StorageProtectionSetupRouter,
+        StrongholdSetupRouter,
+    } from './views'
+    import { OnboardingRoute, onboardingRoute } from '@core/router'
+</script>
+
+{#if $onboardingRoute === OnboardingRoute.AppSetup}
+    <Transition>
```
[the remark this answers is not in the document]
Comment (paul-boegelsack) [reply packages/mobile/routes/onboarding/OnboardingRouter.svelte:15]: I'll remove the transitions with the next PR (mnemonic views and congrats screen) 👍  I updated the Router files and don't want to cause merge conflicts.

File: packages/mobile/components/OnboardingLayout.svelte [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                    icon="arrow-left"
+                    classes={busy ? 'pointer-events-none text-gray-500' : 'cursor-pointer text-blue-500'}
+                />
+            </button>
+        {/if}
+    </header>
+    <!-- TODO: fix flex-col-reverse scrolls mobile-top-content to bottom -->
+    <div class="flex {reverseContent ? 'flex-col-reverse' : 'flex-col'} overflow-y-auto h-full pt-5">
+        <div class={$isKeyboardOpen ? 'h-0' : 'mt-24'}>
+            <slot name="illustration" />
```
[the remark this answers is not in the document]
Comment (paul-boegelsack) [reply packages/mobile/components/OnboardingLayout.svelte]: It is much cleaner without the slots. I just realized the paper plane disappears on welcome before it reaches the edge of the screen ✈️

**Comment by paul-boegelsack** [comment]
> Wohoo amazing work man 🚀🚀 🚀 🚀 🚀 Just added some comments and also waiting for a new update on this PR with the missing views, though I think they could perhaps be added in a different PR, up to you. 💪🏼 💪🏼 💪🏼 💪🏼

Thank you for reviewing 🥳 I'll add the mnemonic views as well as congrats with a different PR. This one is big enough I guess 😅

**Review by begonaalvarezd — APPROVED** [review]
🔥

**Pull request merged**